index.js

```diff
@@ -1,144 +1,166 @@
+ *
+ * Returns a middleware that will validate a token.
+ *
+ * (See: https://tools.ietf.org/html/rfc6749#section-7)
+ */
+
+ExpressOAuthServer.prototype.authenticate = function(options) {
+  const fn = async function(req, res, next) {
+    const request = new Request(req);
+    const response = new Response(res);
+
+    let token
+
+    try {
+      token = await this.server.authenticate(request, response, options);
+    } catch (e) {
+      return handleError.call(this, e, req, res, null, next);
     }
-  }
 
-  /**
-   * Authorization Middleware.
-   *
-   * Returns a middleware that will authorize a client to request tokens.
-   *
-   * (See: https://tools.ietf.org/html/rfc6749#section-3.1)
-   */
-  authorize(options) {
-    return async (req, res, next) => {
-      const request = new Request(req);
-      const response = new Response(res);
-      let code;
-      try {
-        code = await this.server.authorize(request, response, options);
-      } catch (err) {
-        this._handleError(res, response, err, next);
-        return;
-      }
-      res.locals.oauth = { code };
-      if (this.continueMiddleware) {
-        next();
-      }
-      return this._handleResponse(req, res, response);
+    res.locals.oauth = { token };
+    next();
+  };
+  return fn.bind(this);
+};
+
+/**
+ * Authorization Middleware.
+ *
+ * Returns a middleware that will authorize a client to request tokens.
+ *
+ * (See: https://tools.ietf.org/html/rfc6749#section-3.1)
+ */
+
+ExpressOAuthServer.prototype.authorize = function(options) {
+  const fn = async function(req, res, next) {
+    const request = new Request(req);
+    const response = new Response(res);
+
+    let code
+
+    try {
+      code = await this.server.authorize(request, response, options);
+    } catch (e) {
+      return handleError.call(this, e, req, res, response, next);
     }
-  }
 
+    res.locals.oauth = { code: code };
+    if (this.continueMiddleware) {
+      next();
+    }
 
-  /**
-   * Authorization Middleware.
-   *
-   * Returns a middleware that will authorize a client to request tokens.
-   *
-   * (See: https://tools.ietf.org/html/rfc6749#section-3.1)
-   */
-  token(options) {
-    return async (req, res, next) => {
-      const request = new Request(req);
-      const response = new Response(res);
-      let token;
-      try {
-        token = await this.server.token(request, response, options);
-      } catch (err) {
-        this._handleError(res, response, err, next);
-        return;
-      }
-      res.locals.oauth = { token };
-      if (this.continueMiddleware) {
-        next();
-      }
-      return this._handleResponse(req, res, response);
+    return handleResponse.call(this, req, res, response);
+  };
+
+  return fn.bind(this);
+};
+
+/**
+ * Grant Middleware.
+ *
+ * Returns middleware that will grant tokens to valid requests.
+ *
+ * (See: https://tools.ietf.org/html/rfc6749#section-3.2)
+ */
+
+ExpressOAuthServer.prototype.token = function(options) {
```

Comment:
Why isnt this a class method?

Reply:
Oh, you're right they should all be an es6 class. I did this once but reverted the changes and forgot to write as class, again then. Will do.